hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/DNRPCLoadGenerator.java

```diff
@@ -0,0 +1,165 @@
+  private ByteString payloadReqBytes;
+  private int payloadRespSize;
+  private Pipeline pipeline;
+  private ContainerInfo containerInfo;
+  private XceiverClientSpi client;
+  private XceiverClientFactory xceiverClientManager;
+  private ByteString payloadReq;
+  @Option(names = {"--payload-req"},
+          description =
+                  "Specifies the size of payload in KB in RPC request. " +
+                          "Max size is 2097151 KB",
+          defaultValue = "0")
+  private int payloadReqSizeKB = 0;
+
+  @Option(names = {"--clients"},
+      description =
+          "Number of clients, defaults 1.",
+      defaultValue = "1")
+  private int clientsCount = 1;
+
+  @Option(names = {"--payload-resp"},
+          description =
+                  "Specifies the size of payload in KB in RPC response. " +
+                          "Max size is 2097151 KB",
+          defaultValue = "0")
+  private int payloadRespSizeKB = 0;
+
+  @Option(names = {"--ratis"},
+      description = "Write to Ratis log, skip flag for read-only EchoRPC " +
+          "request")
+  private boolean writeToRatis = false;
+  @Option(names = {"--containerID"},
+      description = "Send echo to DataNodes associated with this container")
+  private long containerID;
+
+  @CommandLine.ParentCommand
+  private Freon freon;
+
+  // empy constructor for picocli
+  DNRPCLoadGenerator() {
+  }
+
+  @VisibleForTesting
+  DNRPCLoadGenerator(OzoneConfiguration ozoneConfiguration) {
+    this.configuration = ozoneConfiguration;
+  }
+
+  @Override
+  public Void call() throws Exception {
+    Preconditions.checkArgument(payloadReqSizeKB >= 0,
+            "OM echo request payload size should be positive value or zero.");
+    Preconditions.checkArgument(payloadRespSizeKB >= 0,
+            "OM echo response payload size should be positive value or zero.");
+
+    if (configuration == null) {
+      configuration = freon.createOzoneConfiguration();
+    }
+    scmClient = new ContainerOperationClient(configuration);
+    containerInfo = scmClient.getContainer(containerID);
+
+    List<Pipeline> pipelineList = scmClient.listPipelines();
+    pipeline = pipelineList.stream()
+        .filter(p -> p.getId().equals(containerInfo.getPipelineID()))
+        .findFirst()
+        .orElse(null);
+    xceiverClientManager = new XceiverClientManager(configuration);
+    client = xceiverClientManager.acquireClient(pipeline);
+
+    init();
+    payloadReqBytes = UnsafeByteOperations.unsafeWrap(RandomUtils.nextBytes(
+        calculateMaxPayloadSize(payloadReqSizeKB)));
+    payloadRespSize = calculateMaxPayloadSize(payloadRespSizeKB);
+    timer = getMetrics().timer("rpc-payload");
+    try {
+      runTests(this::sendRPCReq);
+    } finally {
+      xceiverClientManager.releaseClient(client, false);
+      xceiverClientManager.close();
+      scmClient.close();
+    }
+    return null;
+  }
+
+  private int calculateMaxPayloadSize(int payloadSizeKB) {
```

Comment:
I think we can reuse the methods under `org.apache.hadoop.ozone.common.PayloadUtils` for calculating the max payload size here.